cpp/src/arrow/flight/sql/odbc/odbc_api.cc

```diff
@@ -1147,4 +1147,88 @@ SQLRETURN SQLColumns(SQLHSTMT stmt, SQLWCHAR* catalogName, SQLSMALLINT catalogNa
     return SQL_SUCCESS;
   });
 }
+
+SQLRETURN SQLColAttribute(SQLHSTMT stmt, SQLUSMALLINT recordNumber,
+                          SQLUSMALLINT fieldIdentifier, SQLPOINTER characterAttributePtr,
+                          SQLSMALLINT bufferLength, SQLSMALLINT* outputLength,
+                          SQLLEN* numericAttributePtr) {
+  LOG_DEBUG(
+      "SQLColAttributeW called with stmt: {}, recordNumber: {}, "
+      "fieldIdentifier: {}, characterAttributePtr: {}, bufferLength: {}, "
+      "outputLength: {}, numericAttributePtr: {}",
+      stmt, recordNumber, fieldIdentifier, characterAttributePtr, bufferLength,
+      fmt::ptr(outputLength), fmt::ptr(numericAttributePtr));
+  using ODBC::ODBCDescriptor;
+  using ODBC::ODBCStatement;
+  return ODBCStatement::ExecuteWithDiagnostics(stmt, SQL_ERROR, [=]() {
+    ODBCStatement* statement = reinterpret_cast<ODBCStatement*>(stmt);
+    ODBCDescriptor* ird = statement->GetIRD();
+    SQLINTEGER outputLengthInt;
+    switch (fieldIdentifier) {
```

Comment:
SQLColAttribute needs to handle the special SQL_COLUMN_* identifiers. I think these were for ODBC 2.x